pkg/remote/remote.go

```diff
@@ -115,20 +145,40 @@ func NewClient(address string) (Client, error) {
 
 	host := parsedURL.Host
 	pass, _ := parsedURL.User.Password()
-	port := defaultPort
+
+	var port int
 
 	if parsedURL.Port() != "" {
 		// we're safe to ignore error here since `url.Parse` checks if port is valid integer
 		port, _ = strconv.Atoi(parsedURL.Port())
 		host = strings.Split(host, ":")[0]
 	}
 
+	derivedHost := ssh_config.Get(host, "HostName")
```

Comment:
What "derived host" mean and where does it come from? Any documentation out there for it?

Reply:
It's the host defined in the `~/.ssh/config` file for the url's host. The host from the ssh config file overrides the host we connect to.